include/llbuild/Core/BuildDB.h

```diff
@@ -127,8 +128,8 @@ class BuildDB {
   ///
   /// \param keys_out [out] The known keys will be appended to this vector.
   /// \param error_out [out] Error string if return value is false.
-  virtual bool getKeys(std::vector<KeyType>& keys_out, std::string* error_out) = 0;
-
+  virtual bool getKeys(std::map<KeyID, KeyType>& keys_out, std::string* error_out) = 0;
```

Comment:
Related to my comment above, this change should not be made.  The internal database keyIDs should never be returned to the client.